src/Databases/DatabaseAtomic.cpp

```diff
@@ -420,7 +421,46 @@ void DatabaseAtomic::loadStoredObjects(ContextMutablePtr local_context, bool has
 {
     /// Recreate symlinks to table data dirs in case of force restore, because some of them may be broken
     if (has_force_restore_data_flag)
-        fs::remove_all(path_to_table_symlinks);
+    {
+        for (const auto & table_path : fs::directory_iterator(path_to_table_symlinks))
+        {
+            /// Only symlinks should be removed, regular directories will be renamed
+            /// (to avoid accidental data loss)
+            if (fs::is_symlink(table_path))
+            {
+                fs::remove(table_path);
+                continue;
+            }
```

Comment:
It should not contain anything but symlinks. Maybe just throw exception if it contain a regular file?

Reply:
That was my first choice, but then I decided to make it more robust, but since you also suggesting exception, maybe renaming does not worth it. Ok, let's just throw.